The test suite does not build with recent versions of tasty / tasty-hspec

```
tests/Groups.hs:10:26: error:
    Module ‘Test.Tasty.Hspec’ does not export ‘Spec’
   |
10 | import Test.Tasty.Hspec (Spec, testSpec, describe, it, shouldBe)
   |                          ^^^^

tests/Groups.hs:10:42: error:
    Module ‘Test.Tasty.Hspec’ does not export ‘describe’
   |
10 | import Test.Tasty.Hspec (Spec, testSpec, describe, it, shouldBe)
   |                                          ^^^^^^^^

tests/Groups.hs:10:52: error:
    Module ‘Test.Tasty.Hspec’ does not export ‘it’
   |
10 | import Test.Tasty.Hspec (Spec, testSpec, describe, it, shouldBe)
   |                                                    ^^

tests/Groups.hs:10:56: error:
    Module ‘Test.Tasty.Hspec’ does not export ‘shouldBe’
   |
10 | import Test.Tasty.Hspec (Spec, testSpec, describe, it, shouldBe)
   |                                                        ^^^^^^^^
```

--- a/spake2.cabal
+++ b/spake2.cabal
@@ -86,7 +86,7 @@ test-suite tasty
     , protolude >=0.3 && <0.4
     , spake2
     , tasty
-    , hspec >= 1.2 && <2
+    , hspec >= 2.10 && <3
     , tasty-hspec
     , hspec-expectations >= 0.8.2 && <0.9
   other-modules:

--- a/spake2.cabal
+++ b/spake2.cabal
@@ -86,7 +86,9 @@ test-suite tasty
     , protolude >=0.3 && <0.4
     , spake2
     , tasty
+    , hspec >= 1.2, <2
     , tasty-hspec
+    , hspec-expectations >= 0.8.2, <0.9
   other-modules:
       Groups
       Integration

--- a/spake2.cabal
+++ b/spake2.cabal
@@ -86,9 +86,9 @@ test-suite tasty
     , protolude >=0.3 && <0.4
     , spake2
     , tasty
-    , hspec >= 1.2, <2
+    , hspec >= 1.2 && <2
     , tasty-hspec
-    , hspec-expectations >= 0.8.2, <0.9
+    , hspec-expectations >= 0.8.2 && <0.9
   other-modules:
       Groups
       Integration

--- a/tests/Groups.hs
+++ b/tests/Groups.hs
@@ -7,7 +7,10 @@ import Protolude hiding (group)
 import Crypto.Error (CryptoFailable(..))
 import Test.QuickCheck (Gen, (===), arbitrary, forAll, property)
 import Test.Tasty (TestTree)
-import Test.Tasty.Hspec (Spec, testSpec, describe, it, shouldBe)
+import Test.Tasty.Hspec (testSpec)
+
+import Test.Hspec (Spec, describe, it)
+import Test.Hspec.Expectations (shouldBe)
 
 import Crypto.Spake2.Group (AbelianGroup(..), Group(..))
 import Crypto.Spake2.Groups

--- a/tests/Integration.hs
+++ b/tests/Integration.hs
@@ -10,7 +10,11 @@ import qualified Data.ByteString.Char8 as Char8
 import qualified System.IO as IO
 import qualified System.Process as Process
 import Test.Tasty (TestTree)
-import Test.Tasty.Hspec (testSpec, describe, it, shouldBe)
+import Test.Tasty.Hspec (testSpec)
+
+import Test.Hspec (describe, it)
+import Test.Hspec.Expectations (shouldBe)
+
 
 import qualified Crypto.Spake2 as Spake2
 import Crypto.Spake2.Group (Group(arbitraryElement))

--- a/tests/Spake2.hs
+++ b/tests/Spake2.hs
@@ -2,7 +2,11 @@ module Spake2 (tests) where
 
 import Protolude hiding (group)
 import Test.Tasty (TestTree)
-import Test.Tasty.Hspec (testSpec, describe, it, shouldBe, shouldNotBe)
+import Test.Tasty.Hspec (testSpec)
+
+import Test.Hspec (describe, it)
+import Test.Hspec.Expectations (shouldBe, shouldNotBe)
+
 
 import Crypto.Hash (SHA256(..))
 import qualified Crypto.Spake2 as Spake2

